src/minion.c

```diff
@@ -274,7 +274,9 @@ demon_talk(register struct monst *mtmp)
 
     if (uwep && (uwep->oartifact == ART_EXCALIBUR
                  || uwep->oartifact == ART_DEMONBANE)) {
-        pline("%s looks very angry.", Amonnam(mtmp));
+        if (canspotmon(mtmp)) {
+            pline("%s looks very angry.", Amonnam(mtmp));
+        }
```

Comment:
In neither vanilla nor xNetHack I'm getting this message when blind and wielding excalibur. Can you provide steps to reproduce?

Reply:
IIRC I saw it when I climbed up a flight of stairs from below into a demon lair I had bypassed earlier by digging down.  I think my character wasn't blind, but the boss was a few squares away in the dark and was not visible for that reason.

Reply:
<img width="564" alt="Screen Shot 2021-04-30 at 9 17 08 AM" src="https://user-images.githubusercontent.com/40038830/116700558-e78d8e80-a994-11eb-885d-30ccf78389a4.png">

I just reproduced it in vanilla, so maybe this is another one that should be a vanilla PR instead.

Reply:
I moved this one to a vanilla PR as well.